### DISCUSSION TITLE
Pressure is not a registered object

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#25619] [Q&A General]
Hello all,

I am a new user of MOOSE trying to submit my first job. Below is the output I get after send my job. Can anyone help me resolve it.
```
"/cm/local/apps/slurm/var/spool/job1678239/slurm_script: line 17: source: /home/emmanuel324/.moose_profile: is a directory

[31m
*** ERROR ***
A 'Pressure' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.[39m


[31m
*** ERROR ***
A 'Pressure' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.[39m

--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 1 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------"
```

**Comment by GiudGiud** [comment]
Hello

The first error seems to be that `/home/emmanuel324/.moose_profile` is not a file

The second error is that you are using the TensorMechancis Pressure object, but your application does not seem to have access to it
Did you enable Tensor mechanics in the Makefile?

Guillaume

**Reply by Emmanuel-324** [reply]
If you are referring to the modules in my Makefile, yes I turned all of the to yes

**Reply by GiudGiud** [reply]
did you `make clobberall` then recompile after?

**Reply by Emmanuel-324** [reply]
I have a quick question. I get confused sometimes, do I place my job and sbatch file in my application or outside my application?

**Reply by Emmanuel-324** [reply]
Yes I did recompile after adding the .h and .c files

**Reply by GiudGiud** [reply]
The way I do it is:
- create a folder for the run with the input file
- place a symlink to the executable in that folder. It points to the executable in the application folder
- put the job file in that folder and run it from there too

**Reply by Emmanuel-324** [reply]
Sorry but aside creating a folder and putting the job in the folder I do not under your second point or how to do that

**Reply by GiudGiud** [reply]
to create a symlink: `ln -s <folder with the app>/<name of the executable, probably app-opt> ./`

**Reply by Emmanuel-324** [reply]
so my app name is belson and my job is in the folder called "kks_multiphase_largescale_large". Can you help create the command for the symlink. If you need any additional info i can provide.

**Reply by GiudGiud** [reply]
what does `pwd` return in your application folder?

**Reply by Emmanuel-324** [reply]
/home/emmanuel324/projects/belson

**Reply by GiudGiud** [reply]
then go in the folder you want to use for the run and type:
```
ln -s /home/emmanuel324/projects/belson/belson-opt ./
```

**Reply by Emmanuel-324** [reply]
So anytime I want to run my job I need to call this command and go ahead to use the sbatch command?

**Reply by GiudGiud** [reply]
no, you do that and you should see a link in that folder.

Now in the sbatch job you can use that link to the executable to do:
./belson-opt -i input_file.i

**Reply by Emmanuel-324** [reply]
Yeah I can see belson-opt in my job folder.


./belson-opt -i kks_multiphase_largescale_large.i   (what is this command line going to do if you don't mind me asking) ps: kks_multiphase_largescale_large.i is my job file

**Reply by GiudGiud** [reply]
run the executable with the input file.
It seems you should follow the tutorial:
https://mooseframework.inl.gov/workshop/index.html
